File: android/app/src/main/java/com/vidacotidiana/app/core/mock/MockData.kt
package com.vidacotidiana.app.core.mock

import java.time.LocalDate

/**
 * UX-006: mock data for the 6 scaffolding-only modules authorized by the
 * user (Documentos, Inventario, Garantías, Mantenimiento, Suscripciones,
 * Familia) — presentation only, no backend, no persistence. Kept in its own
 * file, referenced only from each module's ViewModel/State, never inlined
 * into a composable, so it's obvious at a glance what's mock vs. real when
 * reading a screen file.
 */

data class MockDocument(val id: String, val name: String, val category: String, val date: String, val sizeLabel: String)
data class MockDocumentCategory(val name: String, val count: Int)

/**
 * UX-007: `expiresOn` is a new structured field, added for Calendario (needs
 * a real `LocalDate` to place a marker on the month grid) — `expiresLabel`
 * only ever had a display string ("12 Mayo 2026"), unlike Web's
 * `mockData.ts`, which already had a structured `expiresAt`. Values below
 * are the same dates `expiresLabel` already spelled out, not new/invented
 * ones.
 */
data class MockWarranty(val id: String, val product: String, val category: String, val expiresLabel: String, val expiresOn: LocalDate, val status: WarrantyStatus)
enum class WarrantyStatus { VIGENTE, POR_VENCER, VENCIDA }

data class MockInventoryItem(val id: String, val name: String, val category: String, val status: String)

/** UX-007: `nextDueOn` — see `MockWarranty.expiresOn`'s comment, same reasoning. */
data class MockMaintenanceRecord(val id: String, val item: String, val task: String, val lastDoneLabel: String, val nextDueLabel: String, val nextDueOn: LocalDate, val status: MaintenanceStatus)
enum class MaintenanceStatus { AL_DIA, PROXIMO, VENCIDO }

data class MockSubscription(val id: String, val name: String, val category: String, val priceLabel: String, val renewsLabel: String)

data class MockFamilyMember(val id: String, val name: String, val relationship: String)

object MockData {
    val documentCategories = listOf(
        MockDocumentCategory("Identificación", 5),
        MockDocumentCategory("Comprobantes", 8),
        MockDocumentCategory("Seguros", 4),
        MockDocumentCategory("Contratos", 3),
        MockDocumentCategory("Otros", 4),
    )

    val documents = listOf(
        MockDocument("doc-1", "INE_David.pdf", "Identificación", "02 May 2024", "1.2 MB"),
        MockDocument("doc-2", "Pasaporte.pdf", "Identificación", "15 Abr 2024", "2.1 MB"),
        MockDocument("doc-3", "Poliza_Seguro_Auto.pdf", "Seguros", "10 Abr 2024", "1.8 MB"),
        MockDocument("doc-4", "Recibo_Luz_Abril.pdf", "Comprobantes", "05 Abr 2024", "1.1 MB"),
    )

    val warranties = listOf(
        MockWarranty("war-1", "Laptop Dell XPS 13", "Electrónicos", "12 Mayo 2026", LocalDate.of(2026, 5, 12), WarrantyStatus.POR_VENCER),
        MockWarranty("war-2", "Lavadora Samsung", "Hogar", "20 Ago 2026", LocalDate.of(2026, 8, 20), WarrantyStatus.VIGENTE),
        MockWarranty("war-3", "Refrigerador LG", "Hogar", "01 Ene 2026", LocalDate.of(2026, 1, 1), WarrantyStatus.VENCIDA),
        MockWarranty("war-4", "Auto Toyota Corolla", "Vehículos", "30 Sep 2027", LocalDate.of(2027, 9, 30), WarrantyStatus.VIGENTE),
    )

    val inventoryCategories = listOf("Todos", "Hogar", "Electrónicos", "Vehículos")

    val inventoryItems = listOf(
        MockInventoryItem("inv-1", "Laptop Dell XPS 13", "Electrónicos", "En uso"),
        MockInventoryItem("inv-2", "Lavadora Samsung", "Hogar", "En uso"),
        MockInventoryItem("inv-3", "Sofá Sala", "Hogar", "En uso"),
        MockInventoryItem("inv-4", "Auto Toyota Corolla", "Vehículos", "En uso"),
    )

    val maintenanceRecords = listOf(
        MockMaintenanceRecord("mnt-1", "Auto Toyota Corolla", "Cambio de aceite", "15 Feb 2026", "15 Ago 2026", LocalDate.of(2026, 8, 15), MaintenanceStatus.PROXIMO),
        MockMaintenanceRecord("mnt-2", "Lavadora Samsung", "Limpieza de filtro", "01 Ene 2026", "01 Jul 2026", LocalDate.of(2026, 7, 1), MaintenanceStatus.AL_DIA),
        MockMaintenanceRecord("mnt-3", "Refrigerador LG", "Revisión general", "10 Nov 2025", "10 May 2026", LocalDate.of(2026, 5, 10), MaintenanceStatus.VENCIDO),
    )

    val subscriptions = listOf(
        MockSubscription("sub-1", "Netflix", "Entretenimiento", "$219/mes", "20 Ago 2026"),
        MockSubscription("sub-2", "Spotify", "Entretenimiento", "$115/mes", "05 Ago 2026"),
        MockSubscription("sub-3", "Almacenamiento en la nube", "Servicios", "$99/mes", "12 Ago 2026"),
    )

    val familyMembers = listOf(
        MockFamilyMember("fam-1", "Ana García", "Esposa"),
        MockFamilyMember("fam-2", "María García", "Hija"),
        MockFamilyMember("fam-3", "Carlos García", "Hijo"),
    )
}

File: web/src/core/mock/mockData.ts
/**
 * UX-006: mock data for the 6 scaffolding-only modules (Documentos,
 * Inventario, Garantías, Mantenimiento, Suscripciones, Familia) — zero
 * business logic, zero endpoints. Values match the reference images'
 * sample data (docs/Ejemplo Web vida Cotidiana .png). Kept in its own
 * module and imported by each page's hook, never inlined into a component —
 * mirrors android/.../core/mock/MockData.kt.
 */

export interface MockDocumentCategory {
  id: string
  name: string
  count: number
}

export interface MockDocument {
  id: string
  name: string
  category: string
  uploadedAt: string
}

export const documentCategories: MockDocumentCategory[] = [
  { id: 'identificacion', name: 'Identificación', count: 5 },
  { id: 'comprobantes', name: 'Comprobantes', count: 8 },
  { id: 'seguros', name: 'Seguros', count: 4 },
  { id: 'contratos', name: 'Contratos', count: 3 },
  { id: 'otros', name: 'Otros', count: 4 },
]

export const documents: MockDocument[] = [
  { id: 'doc-1', name: 'INE_David.pdf', category: 'Identificación', uploadedAt: 'Hace 2 horas' },
  { id: 'doc-2', name: 'Factura Netflix.pdf', category: 'Comprobantes', uploadedAt: 'Hace 5 horas' },
  { id: 'doc-3', name: 'Póliza auto 2026.pdf', category: 'Seguros', uploadedAt: 'Hace 1 día' },
]

export type WarrantyStatus = 'VIGENTE' | 'POR_VENCER' | 'VENCIDA'

export interface MockWarranty {
  id: string
  item: string
  expiresAt: string
  status: WarrantyStatus
}

export const warranties: MockWarranty[] = [
  { id: 'w-1', item: 'Laptop Dell XPS 13', expiresAt: '2027-03-10', status: 'VIGENTE' },
  { id: 'w-2', item: 'Seguro de auto', expiresAt: '2026-05-12', status: 'POR_VENCER' },
  { id: 'w-3', item: 'Lavadora Samsung', expiresAt: '2026-01-20', status: 'VENCIDA' },
]

export interface MockInventoryItem {
  id: string
  name: string
  category: string
  location: string
}

export const inventoryCategories = ['Electrónicos', 'Hogar', 'Vehículos']

export const inventoryItems: MockInventoryItem[] = [
  { id: 'inv-1', name: 'Laptop Dell XPS 13', category: 'Electrónicos', location: 'En uso' },
  { id: 'inv-2', name: 'Lavadora Samsung', category: 'Hogar', location: 'En uso' },
  { id: 'inv-3', name: 'Sofá Sala', category: 'Hogar', location: 'En uso' },
  { id: 'inv-4', name: 'Auto Toyota Corolla', category: 'Vehículos', location: 'En uso' },
]

export type MaintenanceStatus = 'AL_DIA' | 'PROXIMO' | 'VENCIDO'

export interface MockMaintenanceRecord {
  id: string
  item: string
  nextDueAt: string
  status: MaintenanceStatus
}

export const maintenanceRecords: MockMaintenanceRecord[] = [
  { id: 'm-1', item: 'Cambio de aceite — Toyota Corolla', nextDueAt: '2026-09-01', status: 'AL_DIA' },
  { id: 'm-2', item: 'Servicio de auto', nextDueAt: '2026-05-16', status: 'PROXIMO' },
  { id: 'm-3', item: 'Revisión de calentador', nextDueAt: '2026-04-01', status: 'VENCIDO' },
]

export interface MockSubscription {
  id: string
  name: string
  price: string
  cycle: string
}

export const subscriptions: MockSubscription[] = [
  { id: 's-1', name: 'Netflix', price: '$219 MXN', cycle: 'Mensual' },
  { id: 's-2', name: 'Spotify', price: '$115 MXN', cycle: 'Mensual' },
  { id: 's-3', name: 'iCloud+', price: '$29 MXN', cycle: 'Mensual' },
]

export interface MockFamilyMember {
  id: string
  name: string
  relationship: string
}

export const familyMembers: MockFamilyMember[] = [
  { id: 'fam-1', name: 'Ana García', relationship: 'Esposa' },
  { id: 'fam-2', name: 'María García', relationship: 'Hija' },
  { id: 'fam-3', name: 'Carlos García', relationship: 'Hijo' },
]

/**
 * UX-009: Home's "Resumen de actividades" — mock only, on the user's
 * explicit instruction ("genera el mock, luego implementamos la lógica").
 * There is no activity-feed endpoint today (the backend's audit log,
 * BE-029, is server-side/security-only, never exposed to the client) — this
 * is presentation scaffolding for a feature that doesn't exist yet, same
 * spirit as the other 6 mock modules. Every consumer must show these as
 * clearly simulated (subtitle ends in "· simulado"), never indistinguishable
 * from a real event.
 */
export type MockActivityKind = 'task' | 'share' | 'warranty' | 'document'

export interface MockActivity {
  id: string
  kind: MockActivityKind
  text: string
  timeLabel: string
}

export const homeActivities: MockActivity[] = [
  { id: 'act-1', kind: 'task', text: 'Completaste "Pagar servicio de luz"', timeLabel: 'Hace 2 horas' },
  { id: 'act-2', kind: 'share', text: 'Compartiste "Revisión del auto" con Ana', timeLabel: 'Ayer' },
  { id: 'act-3', kind: 'warranty', text: 'Agregaste la garantía de "Lavadora Samsung"', timeLabel: 'Hace 3 días' },
]

/**
 * UX-009: Home's third metric card, "Próximos eventos" — mock (same
 * instruction as above). "Evento" isn't a real concept in the API today
 * (reminders are "tareas", not "eventos") — this stays a static mock number
 * until a real events concept exists, not silently reinterpreted as
 * upcoming reminders.
 */
export const homeUpcomingEventsCount = 2
